fix: access unit from latest day

TheJulianJES opened the pull request:
The current ``unit`` field isn't available anymore. See https://documentation.viessmann.com/static/changelog under ``Removal of standalone "unit"-properties`` for more information.

Fixes https://github.com/somm15/PyViCare/issues/263
Related HA issue: https://github.com/home-assistant/core/issues/77693

https://github.com/somm15/PyViCare/pull/260 already attempts to fix this but seemingly access non-existing fields(?)

This PR always grabs the unit from the latest ``day``.
(I'm not sure if it would be better to grab the unit from ``week``, ``month``, or ``year``. I guess ``day`` should always exist(?))

Note: All dumps still need to be replaced(?)

<details>
  <summary>Click here to see a bit of the new API output</summary>
  
API output for ``devices/0/features/heating.gas.consumption.dhw``:
  
```json
{
    "data": {
        "properties": {
            "day": {
                "type": "array",
                "value": [
                    0,
                    21,
                    23,
                    23,
                    25,
                    23,
                    23,
                    22
                ],
                "unit": "kilowattHour"
            },
            "week": {
                "type": "array",
                "value": [
                    102,
                    153,
                    155,
                    128,
                    90,
                    153,
                    166,
                    181,
                    199,
                    182,
                    199,
                    189,
                    204,
                    201,
                    218,
                    188,
                    203,
                    217,
                    205,
                    204,
                    221,
                    219,
                    235,
                    222,
                    232,
                    232,
                    257,
                    237,
                    252,
                    254,
                    249,
                    251,
                    240,
                    247,
                    229,
                    229,
                    222,
                    235,
                    233,
                    233,
                    249,
                    227,
                    220,
                    207,
                    196,
                    214,
                    221,
                    192,
                    195,
                    189,
                    212,
                    189,
                    188
                ],
                "unit": "kilowattHour"
            },
            "month": {
                "type": "array",
                "value": [
                    31,
                    597,
                    783,
                    831,
                    915,
                    905,
                    1058,
                    983,
                    1076,
                    1016,
                    974,
                    902,
                    840
                ],
                "unit": "kilowattHour"
            },
            "year": {
                "type": "array",
                "value": [
                    7179,
                    11231,
                    11782
                ],
                "unit": "kilowattHour"
            },
            "dayValueReadAt": {
                "type": "string",
                "value": "2022-09-02T22:10:57.328Z"
            },
            "weekValueReadAt": {
                "type": "string",
                "value": "2022-09-02T08:10:38.943Z"
            },
            "monthValueReadAt": {
                "type": "string",
                "value": "2022-09-02T08:10:39.150Z"
            },
            "yearValueReadAt": {
                "type": "string",
                "value": "2022-09-02T08:10:41.839Z"
            }
        },
        "commands": {},
        "apiVersion": 1,
        "uri": "https://api.viessmann.com/iot/v1/equipment/installations/***/gateways/***/devices/0/features/heating.gas.consumption.dhw",
        "gatewayId": "***",
        "feature": "heating.gas.consumption.dhw",
        "timestamp": "2022-09-02T22:10:59.862Z",
        "isEnabled": true,
        "isReady": true,
        "deviceId": "0"
    }
}
```

</details>

Edit: Confirmed this would fix the issue with a ``Vitodens 200-W (B2HB 2,4-19kW)`` and Home Assistant.

TheJulianJES committed:
fix: access unit from latest day

TheJulianJES commented:
FYI @woehrl01 (as you did the other changes)

(And tests are failing because of the outdated dumps)

woehrl01 commented:
@TheJulianJES thank you for the pr. Can you please add a unit test for your device (dump)?

TheJulianJES commented:
Will do. Should all other dumps/tests be removed (temporarily)?

woehrl01 commented:
Just the individual failing tests, please keep the dumps. I will then add additional tests via the other open PR.

Thank you very much!

TheJulianJES committed:
Commented out failing tests

TheJulianJES committed:
Added alternative Vitodens 200-W dump

TheJulianJES committed:
Added WIP test_Vitodens200W_2.py

TheJulianJES commented:
- I've commented out the failing tests for the other dumps (can also completely remove the comments if wanted)
- also added a dump of my heating system (``Vitodens200W_2.json``)
- also added a work in progress test for the alternative Vitodens dump

Two tests from the ``test_TestForMissingProperties.py`` test are failing with the added dump though.
(I'm not familiar with the codebase, so I can't easily fix this right now.) (Feel free to push further changes to my branch/this PR or point me in the right direction if you have time.)

crazyfx1 commented:
It seems like you have to remove this part of the if statement
`and feature['components'] == []:`
in this line
https://github.com/somm15/PyViCare/blob/50e0e584ba57183a6c4bb022164b0a7bb3f3f22a/tests/test_TestForMissingProperties.py#L127

The `components` field was removed by Viessmann some time ago.

TheJulianJES committed:
Removed components check

TheJulianJES committed:
Ignore "new" properties in test

TheJulianJES commented:
Removing that part seems to to make "new" properties occur in the "TestForMissingProperties" test which have to be excluded for now.
I've ignored them in the test now (but again, I don't know if this is really _correct_).

woehrl01 commented:
Thank for your patience @TheJulianJES I'll have a look to get this merged over the weekend.

woehrl01 commented:
Published as `2.17.0` . Thanks @TheJulianJES

cryd-s commented:
cannot get the gas sensor to the energy dashboard - can someone tell me how to fix this?
